src/cljs/vanilla/widgets/basic_widget.cljs

```diff
@@ -23,7 +28,8 @@
   :remove-widget
   (fn-traced [db [_ widget-id]]
     (delete-widget widget-id)
-    (assoc db :widgets (remove #(= (:key %) widget-id) (:widgets db)))))
+    (assoc db :widgets (remove #(= (:key %) widget-id) (:widgets db))
+              :data-sources (dissoc (:data-sources db) (get-data-source @(rf/subscribe [:active-widgets-by-user]) widget-id)))))
```

Comment:
Interesting. Even without reference counting, a "removed" subscription will be resubscribed if another widget is still subscribing. It looks like this happens a part of layout management. We'll want to keep an eye on if this causes any thrashing.

Reply:
You're quick!  Yep was just going to explain that it still worked based on how widgets get sent over if there are multiple widgets with the same data source.  Besides a flash as the new data updates get sent, no noticeable issues.  Might be worth a check once we turn scheduling back on eventually.